### PR TITLE
Pokemons-PR

### PR DESCRIPTION
**Pull request by shirrotem**

### CODE REVIEW

File: src/app/component/pokemons/pokemons.component.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  pokemons: Pokemon[] = [];
+  searchText: string = "";
+  filteredPokemons: Pokemon[] = [];
+  pokemonsSubscription: Subscription | undefined;
+
+  constructor(private router: Router, private pokService: PokemonService) {}
+
+  ngOnInit(): void {
+    this.pokemonsSubscription = this.pokService.getPokemons().subscribe((val: Pokemon[]) => {
+      this.pokemons = val;
```
Comment (sivanbasa93) [inline src/app/component/pokemons/pokemons.component.ts:22]: if this.pokemons without using remove it

File: src/app/services/pokemon.service.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+})
+export class PokemonService {
+  private pokemons: Pokemon[] = [];
+
+  constructor(private http: HttpClient) {}
+
+  getPokemons(): Observable<Pokemon[]> {
+    return this.fetchPokemonList().pipe(
+      switchMap(arrayOfUrls => {
+        const fetchPromises = arrayOfUrls.map((pokemonObj: any) => {
```
Comment (sivanbasa93) [inline src/app/services/pokemon.service.ts]: please change from any to specific type

File: src/app/services/pokemon.service.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    );
+  }
+
+  private fetchPokemonData(url: string): Observable<Pokemon> {
+    return this.http.get(url).pipe(
+      catchError((error: any) => {
+        console.error("An error occurred:", error);
+        return new Observable<Pokemon>();
+      }),
+      map((pokData: any) => {
```
Comment (sivanbasa93) [inline src/app/services/pokemon.service.ts]: please change from any to specific type

File: src/app/component/pokemons/pokemons.component.ts (hunk 1 above)
Comment (shirrotem) [reply src/app/component/pokemons/pokemons.component.ts:22]: I'm using this to keep the state of the original array between filtering to filtering

**Review by sivanbasa93 — APPROVED** [review]

**Pull request merged**